R/tags.R

```diff
@@ -1056,7 +1056,15 @@ extractPreserveChunks <- function(strval) {
     strval <- paste(strval, collapse = "\n")
 
   # matches contains the index of all the start and end markers
-  matches <- gregexpr(pattern, strval, perl = TRUE)[[1]]
+  startmatches <- gregexpr(startmarker, strval, fixed = TRUE)[[1]]
+  endmatches <- gregexpr(endmarker, strval, fixed = TRUE)[[1]]
+  matches <- c(startmatches, endmatches)
+  o <- order(matches)
+  matches <- matches[o]
+  attr(matches, "match.length") <- c(
+    attr(startmatches, "match.length", TRUE),
+    attr(endmatches, "match.length", TRUE)
+  )[o]
   lengths <- attr(matches, "match.length", TRUE)
```

Comment:
```suggestion
  lengths <- c(
    attr(startmatches, "match.length", TRUE),
    attr(endmatches, "match.length", TRUE)
  )[o]
```

Reply:
The attr value isn't used and is only pulled directly out into `lengths` value

Reply:
FWIW, I had considered this, and decided not too in case, at a latter point, someone wants to use `attr(matches, "match.length")`, but I suppose it's not worth the effort